OneCommunityGlobal/HighestGoodNetworkApp
# Edwin - Changed location and function of red x to remove a user from a task

# Description

<img width="553" alt="Screen Shot 2023-06-17 at 8 01 35 PM" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/42288987/7c314647-85e3-4ab2-a8b7-3954e33f642d">

## Related PRS (if any):
This frontend PR is related to the BE PR [#410](https://github.com/OneCommunityGlobal/HGNRest/pull/410)

## Main changes explained:
- Changed location of 'X' and fixed functionality
- Added View and Interact with 'X' Permission

## How to test:

1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. log as Admin/Owner user
4. go to Dashboard -> Team Member Tasks
5. You should be able to see the 'X' mark besides any task and when clicked a modal should pop up to confirm
6. Clicking to 'X' mark should remove the user from that task
7. log as Volunteer user
8. 'X' mark shouldn't be visible 

## Videos/Screenshots of changes:

Admin/Owner account:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/42288987/1bf1840c-5e7e-432f-9ac1-01532a19714e

Volunteer account:

<img width="913" alt="Screen Shot 2023-06-17 at 8 12 56 PM" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/42288987/76bcf806-313f-4d15-9185-3b2662def899">

## Notes:

Code was reused from [849](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/849). Removing a user from a task should only refresh that person or not refresh the Tasks tab at all. Thus, any performance issues would be out of scope for this PR.
